backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/schema/management/impl/ElasticsearchIndexSchemaExportImpl.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.backend.elasticsearch.schema.management.impl;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.OutputStreamWriter;
+import java.lang.invoke.MethodHandles;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import org.hibernate.search.backend.elasticsearch.client.spi.ElasticsearchRequest;
+import org.hibernate.search.backend.elasticsearch.logging.impl.Log;
+import org.hibernate.search.backend.elasticsearch.schema.management.ElasticsearchIndexSchemaExport;
+import org.hibernate.search.util.common.logging.impl.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.stream.JsonWriter;
+
+public class ElasticsearchIndexSchemaExportImpl implements ElasticsearchIndexSchemaExport {
+	private static final Log log = LoggerFactory.make( Log.class, MethodHandles.lookup() );
+
+	private final Gson userFacingGson;
+	private final String indexName;
+	private final ElasticsearchRequest request;
+
+	public ElasticsearchIndexSchemaExportImpl(Gson userFacingGson, String indexName,
+			ElasticsearchRequest request) {
+		this.userFacingGson = userFacingGson;
+		this.indexName = indexName;
+		this.request = request;
+	}
+
+	@Override
+	public void toFiles(Path targetDirectory) {
+		JsonObject queryParams = new JsonObject();
+		for ( Map.Entry<String, String> entry : request.parameters().entrySet() ) {
+			queryParams.addProperty( entry.getKey(), entry.getValue() );
+		}
+		List<JsonObject> parts = request.bodyParts();
+		JsonElement body = parts.size() == 1 ?
+				parts.get( 0 ) :
+				parts.stream().collect( JsonArray::new, JsonArray::add, JsonArray::addAll );
+
+		write( targetDirectory, "create-index.json", body );
+		write( targetDirectory, "query-params.json", queryParams );
```

Comment:
Ah, you moved the query params out of `create-index.json`? Then maybe call that `create-index-query-params.json`? Otherwise we'll be in trouble if we add other requests (e.g. `drop-index.json`) in the future.

```suggestion
		write( targetDirectory, "create-index.json", body );
		write( targetDirectory, "create-index-query-params.json", queryParams );
```